File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
       
        if len(nums) == 1 and target == 0:
            return 0
        
        if len(nums) == 1 and target == 1:
            return 0

        answer = 0
        for i in range(len(nums)):
            if target <= nums[i] :
                answer =  i
                return answer
           
        if answer == 0:
            return len(nums)
        return answer

        


        

File: 0035-search-insert-position/0035-search-insert-position.cpp
static const bool Booster = [](){
    #pragma GCC optimize("OFast")
    std::ios_base::sync_with_stdio(0);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return true;
}();

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size()-1;
        while(left <= right){
            int middle = left + (right - left ) / 2;
            if(nums[middle] == target){
                return middle;
            }else if(nums[middle] > target){
                right = middle - 1;
            }else if (nums[middle] < target){
                left = middle + 1;
            }
        }
        return left;
    }
};